idb/postgres/internal/encoding/encode.go

```diff
@@ -63,13 +128,122 @@ func EncodeSignedTxnWithAD(stxn types.SignedTxnWithAD) []byte {
 	return EncodeJSON(convertSignedTxnWithAD(stxn))
 }
 
-func init() {
-	jsonCodecHandle = new(codec.JsonHandle)
-	jsonCodecHandle.ErrorIfNoField = true
-	jsonCodecHandle.ErrorIfNoArrayExpand = true
-	jsonCodecHandle.Canonical = true
-	jsonCodecHandle.RecursiveEmptyCheck = true
-	jsonCodecHandle.HTMLCharsAsIs = true
-	jsonCodecHandle.Indent = 0
-	jsonCodecHandle.MapKeyAsString = true
+func nothingToEscape(x string) bool {
+	strBytes := []byte(x)
+	i := 0
+	// Check each rune to see if base64 encoding is needed
+	for i < len(strBytes) {
+		c, csize := utf8.DecodeRune(strBytes[i:])
+		if c == utf8.RuneError {
+			return false
+		}
+		if !unicode.IsPrint(c) {
+			return false
+		}
+		if c == '\\' {
+			return false
+		}
+		i += csize
+	}
+	return true
+}
+
+func EscapeNulls(x string) string {
+	if nothingToEscape(x) {
+		return x
+	}
+
+	xb := []byte(x)
+
+	escapenull := []byte("\\u0000")
+	var out strings.Builder
```

Comment:
This is worse. The prior implementation will produce less memory garbage by doing exactly one allocation of exactly the right size. Why replace the better implementation with a worse implementation? (I know, readability, but the better implementation already existed so moving backwards is frustrating.)